### ISSUE TITLE
Library Mode not working

### ISSUE THREAD

**Issue by suterma** [suterma/vue-peaks#1]
When building for the "Library Mode" the resulting output seems not to be directly usable by another project. 

- Output Javascript seems to be not usable (can not be imported)
- When using the .Vue SFC component directly, an error about mounting arises:

> n]: onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.

**Comment by suterma** [comment]
See this example as an inspiration: https://www.npmjs.com/package/vue-npm-library-example

**Comment by suterma** [comment]
With the above package, I did not succeed yet, there are multiple issues:

- the types are not built into the output dir (dist/ViewerPlugin.d.ts does not get created by the build). I crafted and put them into the dist manually for testing purposes, this workaround is functional
- When using from another project, the browser complains about lifecycle management: `onMounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.`
